src/reducers/eventsReducer.js

```diff
@@ -0,0 +1,75 @@
+import { SET_EVENT_PROFILE, LOAD_EVENTS_LIST_SUCCESS, SET_EVENT_PROFILE_BY_ID, FILTER_EVENTS, SET_EVENTS_FILTER } from '../actions/eventActions';
+
+const initialState = {
+  eventsList: [],
+  filteredEvents: [],
+  eventsFilter: [],
+  eventProfile: {},
+};
+
+const noEventFound = {
+  name: 'No event found',
+};
+
+const topics = {
+  js: ['Javascript', 'JS'],
+  react: ['React', 'ReactJS'],
+  css: ['CSS', 'Cascading Style Sheets'],
+  html: ['HTML', 'Hypertext Markup Language'],
+};
+
+function eventReducer(state = initialState, action) {
+  let events = '';
+  let filteredById = '';
+  const topicsArray = [];
+  switch (action.type) {
+    case LOAD_EVENTS_LIST_SUCCESS:
+      if (state.eventsFilter.length > 0) {
```

Comment:
So, now you're getting events and also filtering them. Don't you find it opposite to single responsibility principle?